salt/netapi/rest_cherrypy/app.py

```diff
@@ -1427,6 +1448,14 @@ def POST(self, **kwargs):
         else:
             creds = cherrypy.serving.request.lowstate
 
+        # Authorization header
+        remote_user = cherrypy.request.headers.get('X-Remote-User')
+        sharedsecret =  cherrypy.request.headers.get('X-Shared-Secret')
+        if remote_user and sharedsecret:
+            creds['username'] = username
+            creds['sharedsecret'] = sharedsecret
+            creds['eauth'] = 'sharedsecret'
+
```

Comment:
We'll see if we can get a working version with the Authorization header (which is not yet read by app.py). One of the issues that we have with this approach is that the eauth information is still in the json data which is difficult to manipulate by the web server. That means that the a client such as pepper would need to figure out it has to do kerberos auth with the frontal but say eauth: "sharedsecret" in the post data, which seems like a hack.